### ISSUE TITLE
better support for long lines (abbreviated, wrapping, etc)

### ISSUE THREAD

**Issue by op2786** [Vector35/binaryninja-api#3289]
Let's say I have a line like this:

`sprintf_s(Buffer, 512, "Hello! Result=%d, Result2=%d, Result3=%d", LongFunctionName1(), LongFunctionName2(), LongFunctionName3());`

I think this line and other lines long like this should be shown like this:

```
sprintf_s(Buffer,
    512,
    "Hello! Result=%d, Result2=%d, Result3=%d",
    LongFunctionName1(),
    LongFunctionName2(),
    LongFunctionName3());
```

Little things like those makes reading decompiler output way more easier.

**Comment by alexrp** [comment]
Those lines are identical?

(That said, totally agree that some lines in the output can get unreasonably long.)

**Comment by op2786** [comment]
Sorry, I fixed it now.

**Comment by jborean93** [comment]
It would also be great if this could also apply to function signatures and not just the inline code. For example instead of

```c
int64_t SpAcquireCredentialsHandleW(LPWSTR pszPrincipal, LPWSTR pszPackage, uint32_t fCredentialUse, PLUID pvLogonID, struct CREDSSP_CRED* pAuthData,  SEC_GET_KEY_FN pGetKeyFn, PVOID pvGetKeyArgument, struct SecHandle* phCredential, int64_t* ptsExpiry)
```

It would could be configured to show

```c
int64_t SpAcquireCredentialsHandleW(
    LPWSTR pszPrincipal,
    LPWSTR pszPackage,
    uint32_t fCredentialUse,
    PLUID pvLogonID,
    struct CREDSSP_CRED* pAuthData, 
    SEC_GET_KEY_FN pGetKeyFn,
    PVOID pvGetKeyArgument,
    struct SecHandle* phCredential,
    int64_t* ptsExpiry)
```

**Comment by alexrp** [comment]
FWIW, as BN is gaining more and more sidebar widgets (especially now with Sidekick) and the right sidebar is actually fully usable, I think this issue is becoming more pressing.

Signatures like this are just not very readable:

![image](https://github.com/Vector35/binaryninja-api/assets/44076/ae65aeb1-2a28-4803-a00e-9cb7d5b264ed)

As the previous comment suggests, this would be much better:

```c
void S1DataQueryStep::NarrowSearchSpace(
    struct S1DataQueryStep* this,
    int32_t keyIndex, 
    uint32_t nodeNameIndex,
    struct S1DataNodeIterator* children,
    int32_t* start,
    int32_t* count, 
    struct S1DataQueryContext* context)
```

(Besides that, an option to not render the `struct`/`union`/`enum` tags in the linear view would also go a long way in saving precious screen real estate. I wonder if there's an existing issue tracking this?)

**Comment by psifertex** [comment]
I agree about the increasing need. We tentatively have this scheduled for the next stable accordingly--though take that with a bit of a grain of salt as we are often overly optimistic early in the dev cycle for a release.

**Comment by psifertex** [comment]
Related issue (might be a duplicate other than the fact that it is referencing strings specifically):

https://github.com/Vector35/binaryninja-api/issues/2366

**Comment by psifertex** [comment]
The general issue has been resolved in builds >= 4.3.6733. We are tracking the remaining request of a non-default setting to add new lines for each parameter in #6472.
